New defaults for SCA tools

martofeld opened the pull request:
Change the default rules for all the tools. The changes can be seen by taking a look at the changed files

jsotuyod committed:
Move chekstyle suppressions to github

jsotuyod committed:
Fine tune checkstyle suppressions

martofeld committed:
Modify the defaults

The new defaults have been defined with @barriosnahuel, @nicous and @jsotuyod

martofeld committed:
Merge branch 'development' of https://github.com/martofeld/static-code-analysis-plugin into development

martofeld committed:
Lint:
 - Ignore dynamic resources (possibly sent from backend) :raised_hands:

Checkstyle
 - Close opened tags :bomb:

 Findbugs
  - Ignore MISSING_TO_STRING_OVERRIDE in activities :+1:

martofeld committed:
Merge branch 'development' into development

jsotuyod commented on defaults/checkstyle/checkstyle-cache.xml:
```diff
@@ -1,83 +1,133 @@
+        </module>
+        <module name="IllegalImport">
+            <property name="severity" value="error"/>
+            <property name="illegalPkgs" value="sun, android.util.Log, framework.junit, javax.annotation"/>
```
esto es muy fuerte para el default... SCA aplica no sólo a proyectos Android, sino también Java, Groovy, Scala... prohibir `javax.annotation` es contradictorio con tener el check de findbugs de nullness (sólo deshabilitado en proyectos Android). Prohubir `android.util.Log` es opinionated, pero en favor de Timber o similares puedo bancarmelo.

Testeaste que puedas poner un full class name y lo reconozca en vez de un package name?

martofeld replied:
Lo probe, no funciono y colgue de sacarlo. javax.annotation se podra agregar solo para Android?

martofeld committed:
Fixes:

Checkstyle:
- Remove invalid package from the invalid packages rule

Findbugs
- Disable the LOST_STACK_TRACE rule since PMD has the same and is faster

martofeld committed:
Merge branch 'development' of https://github.com/martofeld/static-code-analysis-plugin into development

martofeld committed:
Update gitignore to ignore DS_STORE

martofeld committed:
Ignore another fb rule in favor of PMD

martofeld committed:
Reformat android-lint file

jsotuyod commented:
cuando me confirmen esto lo mando a staging y empieza a aplicar para todos. Los cambios (sobre todo de checkstyle), significa que algunos repos van a empezar a romper. Por eso quiero confirmación de @martofeld @nicous @barriosnahuel previo a hacerlo... ya hablaron con los otros teams?

barriosnahuel commented:
No convendría cambiar el major version de SCA? Estamos cambiando el comportamiento completo de una herramienta. Capaz podríamos hacerlo así para que no se les rompa de golpe y puedan priorizar la tarea... qué opinan?
